tests/test_pin_job.py

```diff
@@ -43,3 +43,20 @@ def test_pin_all_jobs(base_url, selenium):
     page = TreeherderPage(selenium, base_url).open()
     page.result_sets[0].pin_all_jobs()
     assert 0 < len(page.pinboard.jobs) <= 500
+
+
+def test_pin_a_bug(base_url, selenium, new_user):
+    """Open treeherder, log in, select unclassified job, pin job, add a bug, save and verify"""
+    page = TreeherderPage(selenium, base_url).open()
+    page.header.login(new_user['email'], new_user['password'])
```

Comment:
I'm seeing a failure where the bidpom/Persona-login window hasn't-yet loaded - can a conditional wait be added to make this more robust?

`________________________________ test_pin_a_bug ________________________________
[gw7] linux2 -- Python 2.7.9 /var/lib/jenkins/jobs/treeherder.adhoc/workspace/.venv/bin/python2.7
base_url = 'https://treeherder.allizom.org'
selenium = <selenium.webdriver.remote.webdriver.WebDriver (session="84d3eeeefa2c4b0ea063b5c84ffa7ff0")>
new_user = {'email': 'charlie633686@personatestuser.org', 'name': 'charlie633686', 'password': 'jIUNgVygizhS4hVA', 'url': 'http://www.mozilla.org/', ...}

    def test_pin_a_bug(base_url, selenium, new_user):
        """Open treeherder, log in, select unclassified job, pin job, add a bug, save and verify"""
        page = TreeherderPage(selenium, base_url).open()
>       page.header.login(new_user['email'], new_user['password'])

tests/test_pin_job.py:51: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pages/base.py:35: in login
    browser_id.sign_in(email, password)
.venv/lib/python2.7/site-packages/bidpom/bidpom.py:20: in sign_in
    sign_in = SignIn(self.selenium, self.timeout)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <bidpom.pages.sign_in.SignIn object at 0x7faa7a130510>
selenium = <selenium.webdriver.remote.webdriver.WebDriver (session="84d3eeeefa2c4b0ea063b5c84ffa7ff0")>
timeout = 10

    def __init__(self, selenium, timeout=10):
        super(SignIn, self).__init__(selenium, timeout)
    
        if not self.selenium.title == self._page_title:
            for handle in self.selenium.window_handles:
                self.selenium.switch_to_window(handle)
                self.wait.until(lambda s: s.title)
                if self.selenium.title == self._page_title:
                    self._sign_in_window_handle = handle
                    break
            else:
>               raise Exception('Popup has not loaded')
E               Exception: Popup has not loaded

.venv/lib/python2.7/site-packages/bidpom/pages/sign_in.py:50: Exception
------------------------------- pytest-selenium --------------------------------
URL: https://treeherder.allizom.org/admin.html#/profiles
SauceLabs Job: http://saucelabs.com/jobs/84d3eeeefa2c4b0ea063b5c84ffa7ff0
________________________ test_that_a_new_user_can_login ________________________
[gw4] linux2 -- Python 2.7.9 /var/lib/jenkins/jobs/treeherder.adhoc/workspace/.venv/bin/python2.7
base_url = 'https://treeherder.allizom.org'
selenium = <selenium.webdriver.remote.webdriver.WebDriver (session="dca598fcea74467382f170fa185298af")>
new_user = {'email': 'roy633687@personatestuser.org', 'name': 'roy633687', 'password': 'Dm9n5qEOUHjVFBgr', 'url': 'http://www.mozilla.org/', ...}

    @pytest.mark.nondestructive
    def test_that_a_new_user_can_login(base_url, selenium, new_user):
        page = TreeherderPage(selenium, base_url).open()
>       page.header.login(new_user['email'], new_user['password'])

tests/test_login.py:13: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pages/base.py:35: in login
    browser_id.sign_in(email, password)
.venv/lib/python2.7/site-packages/bidpom/bidpom.py:20: in sign_in
    sign_in = SignIn(self.selenium, self.timeout)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <bidpom.pages.sign_in.SignIn object at 0x7f43e8b4dc50>
selenium = <selenium.webdriver.remote.webdriver.WebDriver (session="dca598fcea74467382f170fa185298af")>
timeout = 10

    def __init__(self, selenium, timeout=10):
        super(SignIn, self).__init__(selenium, timeout)
    
        if not self.selenium.title == self._page_title:
            for handle in self.selenium.window_handles:
                self.selenium.switch_to_window(handle)
                self.wait.until(lambda s: s.title)
                if self.selenium.title == self._page_title:
                    self._sign_in_window_handle = handle
                    break
            else:
>               raise Exception('Popup has not loaded')
E               Exception: Popup has not loaded

.venv/lib/python2.7/site-packages/bidpom/pages/sign_in.py:50: Exception`